sdk/samples/iot/hub/src/paho_iot_hub_pnp_example.c

```diff
@@ -0,0 +1,744 @@
+
+  // Initialize the hub client with the hub host endpoint and the default connection options
+  AZ_RETURN_IF_FAILED(az_iot_hub_client_init(
+      &client,
+      az_span_slice(iot_hub_hostname_span, 0, (int32_t)strlen(iot_hub_hostname)),
+      az_span_slice(device_id_span, 0, (int32_t)strlen(device_id)),
+      NULL));
+
+  return AZ_OK;
+}
+
+static az_result report_method(az_span payload, az_span response, az_span* out_response)
+{
+  az_result result;
+
+  // Parse the "since" field in the payload
+  az_json_parser jp;
+  az_json_token_member tm;
+  AZ_RETURN_IF_FAILED(az_json_parser_init(&jp, payload));
+  AZ_RETURN_IF_FAILED(az_json_parser_parse_token(&jp, &tm.token));
+  if (tm.token.kind != AZ_JSON_TOKEN_BEGIN_OBJECT)
+  {
+    return AZ_ERROR_PARSER_UNEXPECTED_CHAR;
+  }
+
+  az_span start_time_span = AZ_SPAN_NULL;
+  while (az_succeeded(az_json_parser_parse_token_member(&jp, &tm)))
+  {
+    if (az_span_is_content_equal(method_payload_value, tm.name))
+    {
+      AZ_RETURN_IF_FAILED(az_json_token_get_string(&tm.token, &start_time_span));
+    }
+
+    // else ignore token.
+  }
+
+  // Get the current time as a string
+  time_t rawtime;
+  struct tm* timeinfo;
+  time(&rawtime);
+  timeinfo = localtime(&rawtime);
+  size_t len = strftime(end_time_buffer, sizeof(end_time_buffer), "%Y-%m-%dT%H:%M:%S%z", timeinfo);
+  az_span time_span = az_span_init((uint8_t*)end_time_buffer, (int32_t)len);
+
+  // Build the method response payload
+  az_json_builder json_builder;
+  result = az_json_builder_init(&json_builder, response, NULL);
+  result = az_json_builder_append_begin_object(&json_builder);
+  result = az_json_builder_append_property_name(&json_builder, report_max_temp_name_span);
+  result = az_json_builder_append_int32_number(&json_builder, (int32_t)device_max_temp);
+  result = az_json_builder_append_property_name(&json_builder, report_min_temp_name_span);
+  result = az_json_builder_append_int32_number(&json_builder, (int32_t)device_min_temp);
+  result = az_json_builder_append_property_name(&json_builder, report_avg_temp_name_span);
+  result = az_json_builder_append_int32_number(&json_builder, (int32_t)device_avg_temp);
+  result = az_json_builder_append_property_name(&json_builder, report_start_time_name_span);
+
+  // If the user specified a time, use that as the start | Otherwise use boot time
+  if (az_span_size(start_time_span) > 0)
+  {
+    result = az_json_builder_append_string(&json_builder, start_time_span);
+  }
+  else
+  {
+    result = az_json_builder_append_string(&json_builder, boot_time_span);
+  }
+
+  result = az_json_builder_append_property_name(&json_builder, report_end_time_name_span);
+  result = az_json_builder_append_string(&json_builder, time_span);
+  result = az_json_builder_append_end_object(&json_builder);
+
+  *out_response = az_json_builder_get_json(&json_builder);
+
+  return result;
+}
+
+static int send_method_response(
```

Comment:
humm received_method_response?

Reply:
In the flow here:
```c
az_result response
    = report_method(method_payload_span, method_response_span, &method_response_span);
(void)response;

// Send method response with report as JSON payload
int rc;
if ((rc = send_method_response(&method_request, 200, method_response_span)) != 0)
{
   printf("Unable to send %d response, status %d\n", 200, rc);
}
```
I think it works as is to understand what it's doing. I can change if you don't think so.